### PR TITLE
PP-499: Upgrade RelaySelectionManager

### PR DESCRIPTION
**Pull request by AndresQuijano**
## What

- Adapts the UpgradeSelectionManager to the new version

## Why

- Modernization of the project

## Refs

- [PP-499](https://rsklabs.atlassian.net/browse/PP-499)

### CODE REVIEW

File: src/api/common/HttpClient.ts [hunk 1]
```diff
@@ -70,7 +71,25 @@ export default class HttpClient {
...
+    const pingResponse = await this._httpWrapper.sendPromise<PingResponse>(
+        relayUrl + '/getaddr' + verifierSuffix
+    );
+    if (pingResponse == null) {
+        throw new Error('Relay responded without a body');
+    }
+    log.info(`pingResponse: ${JSON.stringify(pingResponse)}`);
+
+    return pingResponse;
+  }
```
Comment (jurajpiar) [inline src/api/common/HttpClient.ts:89]: this is already implemented in [getChainInfo](https://github.com/rsksmart/rif-relay-client/pull/49/files#diff-17fc46d2093dfe0332d5af3ff266138a5ea4f1bb0f2cb7afa316fb2a122fb1fdL30-L39) isn't it?

File: src/api/common/HttpClient.ts [hunk 2]
```diff
@@ -70,7 +71,25 @@ export default class HttpClient {
...
 }
 
+export const defaultHttpClient = new HttpClient(new HttpWrapper());
```
Comment (jurajpiar) [inline src/api/common/HttpClient.ts:92]:
```suggestion
export const getDefaultHttpClient = (
    opts: AxiosRequestConfig = {},
    loglLevel: LogLevelDesc = 'error'
  ) => new HttpClient(new HttpWrapper(opts, logLevel));
```

though I'd also consider using a static method allowing the use of singleton.

File: src/utils.ts [hunk 3]
```diff
@@ -18,3 +18,102 @@ export const isDeployTransaction = (
...
+  relayWorkerAddress: string;
+  relayManagerAddress: string;
+  relayHubAddress: string;
+  feesReceiver: string;
+  minGasPrice: string;
+  networkId?: string;
+  chainId?: string;
+  ready: boolean;
+  version: string;
+}
```
Comment (jurajpiar) [inline src/utils.ts:32]: Already done as [HubInfo](https://github.com/rsksmart/rif-relay-client/blob/43d3a469d3367c4c4b66be3bf7eb66f5c562f214/src/common/relay.types.ts#L15)

**Review by jurajpiar — APPROVED** [review]
LG, I'd just check the utils as they might be implementing already existing code.

File: src/api/common/HttpClient.ts (hunk 1 above)
Comment (AndresQuijano) [reply src/api/common/HttpClient.ts:89]: Yes, you're right. Done.

File: src/utils.ts (hunk 3 above)
Comment (AndresQuijano) [reply src/utils.ts:32]: Done.

File: src/api/common/HttpClient.ts (hunk 2 above)
Comment (AndresQuijano) [reply src/api/common/HttpClient.ts:92]: I'll use the approach you wrote. I have never be a pro singleton pattern in JS.

File: src/KnownRelaysManager.ts [hunk 4]
```diff
@@ -2,11 +2,11 @@ import log from 'loglevel';
 import type {
```
Comment (franciscotobar) [inline src/KnownRelaysManager.ts:2]: This file needs to be deleted

Comment (AndresQuijano) [reply src/KnownRelaysManager.ts]: Done.

**Review by franciscotobar — APPROVED** [review]
LGTM

**Review by antomor — CHANGES_REQUESTED** [review]
@AndresQuijano Thank you for the changes.
Apart from the comments I left, I'd change the name of the `RelaySelectionManager` to use the function name

**Review by antomor — APPROVED** [review]

**Pull request merged**